src/melee.cpp

```diff
@@ -1369,14 +1393,6 @@ bool player::block_hit(Creature *source, body_part &bp_hit, damage_instance &dam
     // So that we don't suddenly switch that for any reason.
     const bool weapon_blocking = can_weapon_block();
```

Comment:
Now if you did that change in `best_shield`, just changing this to
```
const bool weapon_blocking = !shield.is_null();
```

Should be enough.